### ISSUE TITLE
Missing checks for CKV_DOCKER_3 (User exists)

### ISSUE THREAD

**Issue by adberger** [bridgecrewio/checkov#2218]
**Describe the issue**
https://github.com/bridgecrewio/checkov/blob/master/checkov/dockerfile/checks/UserExists.py doesn't cover all of https://docs.bridgecrew.io/docs/ensure-that-a-user-for-the-container-has-been-created

Bridgecrew Policy ID: BC_DKR_3
Checkov Check ID: CKV_DOCKER_3

ToDo: Additionally check if `gosu` is executed in either `CMD` or `ENTRYPOINT`

**Examples**
```
FROM alpine:3.14
ENTRYPOINT ["gosu", "app", "tini", "--", "/app/run.sh" ]
```

should be passing but isn't:
```
Check: CKV_DOCKER_3: "Ensure that a user for the container has been created"
	FAILED for resource: /Dockerfile.
	File: /Dockerfile:1-50
	Guide: https://docs.bridgecrew.io/docs/ensure-that-a-user-for-the-container-has-been-created
```

**Version (please complete the following information):**
 - Checkov Version 2.0.692

**Comment by nimrodkor** [comment]
Hey @adberger !

This is an interesting one. How would checkov know if the first keyword (i.e. `gosu`) is the name of the command or a user?

Perhaps this should pass:
```Dockerfile
```

**Comment by adberger** [comment]
@nimrodkor I would rather check if `gosu` is **in** `CMD` or `ENTRYPOINT`, despite the location (first, middle, last etc.).

`USER gosu` would not be correct, because `gosu` is the command and `app` the user which gets created by `gosu` (AFAIK)

**Comment by nimrodkor** [comment]
@adberger Right. However, how would you know `gosu` is a user and not the app / executable / command?

**Comment by adberger** [comment]
> @adberger Right. However, how would you know `gosu` is a user and not the app / executable / command?

You could parse it according to the [gosu usage](https://github.com/tianon/gosu#gosu):
`Usage: ./gosu user-spec command [args]`

If `gosu` and then a `user-spec` is set, the test should be valid.

**Comment by nimrodkor** [comment]
I think I got it now. Want to fix the check and open a PR? We'd love the contribution @adberger !

**Comment by adberger** [comment]
@nimrodkor I prepared some changes in my forked repository, which is currently under review by my colleagues:
https://github.com/adberger/checkov

**Comment by tspearconquest** [comment]
Please don't let this be closed.

**Comment by tspearconquest** [comment]
My mistake, please close it.

**Comment by tspearconquest** [comment]
Not stale

**Comment by ArjunMenon-bit** [comment]
Not stale.
Issue fixed in [#2294](https://github.com/bridgecrewio/checkov/pull/2294)
